packages/flutter/lib/src/widgets/debug.dart

```diff
@@ -242,13 +242,33 @@ bool debugCheckHasMediaQuery(BuildContext context) {
 /// assert(debugCheckHasDirectionality(context));
 /// ```
 ///
+/// To improve the error messages you can add some extra color using the
+/// named arguments.
+///
+///  * why: explain why the direction is needed, for example "to resolve
+///    the 'alignment' argument". Should be an adverb phrase describing why.
+///  * hint: explain why this might be happening, for example "The default
+///    value of the 'aligment' argument of the $runtimeType widget is an
+///    AlignmentDirectional value.". Should be a fully punctuated sentence.
```

Comment:
ultra nit, and only if we're using American punctuation rules (personally, I think it's a dumb rule, but it is what it is):
```suggestion
///    AlignmentDirectional value." Should be a fully punctuated sentence.
```

Reply:
The string here is a literal, not a quote, so I'm inclined to go with programming punctuation rather than usual prose punctuation.